src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py

```diff
@@ -2982,3 +2982,89 @@ def __init__(self, name, group, location):
                  checks=[
                      JMESPathCheck('length(@)', 0)
                  ])
+
+
+class SqlVirtualClusterMgmtScenarioTest(ScenarioTest):
+
+    def test_sql_virtual_cluster_mgmt(self):
+        managed_instance_name = self.create_random_name(managed_instance_name_prefix, managed_instance_name_max_length)
+        admin_login = 'admin123'
+        admin_password = 'SecretPassword123'
+
+        is_playback = os.path.exists(self.recording_file)
+        if is_playback:
+            subnet = '/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/RG_MIPlayground/providers/Microsoft.Network/virtualNetworks/VNET_MIPlayground/subnets/VCReservedSubnet1'
+        else:
+            subnet = '/subscriptions/a8c9a924-06c0-4bde-9788-e7b1370969e1/resourceGroups/RG_MIPlayground/providers/Microsoft.Network/virtualNetworks/VNET_MIPlayground/subnets/VCReservedSubnet1'
+
+        license_type = 'LicenseIncluded'
+        loc = 'eastus'
+        v_cores = 8
+        storage_size_in_gb = '32'
+        edition = 'GeneralPurpose'
+        family = 'Gen5'
+        resource_group = "RG_MIPlayground"
```

Comment:
Why are you hardcoding the rg name instead of using @ResourceGroupPreparer ?

Reply:
I used existing subnet and empty virtual cluster (belonging to this RG) to speed up the test - otherwise it would take several hours to create. Then I just reused this resource group for MI, as this test is just focusing on virtual cluster commands. Let me know if this is ok, given the required time to provision new virtual cluster.

Reply:
It is really really important for scenario tests to be fully automated. Anyone other than you (i.e. not even in MI or SQL team) might need to re-record this test in the future, and manual setup steps are extremely painful. CLI team, please share your guidance if the test cannot be automated.

Reply:
Got it, I'll try to modify the test. There are some tricky parts regarding subnet preparation, I'll reach out to CLI team if I get stuck.

Reply:
@jaredmoo sorry for the delay, I was on-call for a week :) Please review.

Reply:
Thank you, it's a huge improvement that this is automated.